js/index.js

```diff
@@ -1,42 +1,107 @@
 // ITERATION 1
 
 function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
+  const price = product.querySelector(".price span"); 
+  const quantity = product.querySelector(".quantity input"); 
+  const subtotal = product.querySelector(".subtotal span");
 
-  //... your code goes here
+  const priceNum = price.innerHTML; 
+  const quantityNum = quantity.value; 
+  const subtotalNum = priceNum * quantityNum;
+  subtotal.innerHTML = subtotalNum; 
+
+  return subtotalNum; 
 }
 
 function calculateAll() {
   // code in the following two lines is added just for testing purposes.
   // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
+  //const singleProduct = document.querySelector('.product');
+  //updateSubtotal(singleProduct);
   // end of test
-
+  let totalSum = 0; 
   // ITERATION 2
   //... your code goes here
+  const products = document.querySelectorAll(".product"); 
+  products.forEach((singleProduct) => {
+    subtotal = updateSubtotal(singleProduct); 
+    totalSum += subtotal 
+  });
+
+
+
 
   // ITERATION 3
   //... your code goes here
+  const total = document.querySelector("#total-value span"); 
+  total.innerHTML = totalSum; 
 }
 
 // ITERATION 4
 
 function removeProduct(event) {
-  const target = event.currentTarget;
-  console.log('The target in remove is:', target);
+  const target = event.currentTarget.parentNode.parentNode; ;
+  //console.log('The target in remove is:', target);
   //... your code goes here
+  target.parentNode.removeChild(target);
+
+  const total = document.querySelector("#total-value span"); 
+  const subtotal = target.querySelector(".subtotal span");
+  
+  total.innerHTML = total.innerHTML - subtotal.innerHTML; 
 }
 
 // ITERATION 5
 
 function createProduct() {
   //... your code goes here
+  let nameElement = document.querySelector
+  (
+    '.create-product input[type="text"]'
+  );
+  let priceElement = document.querySelector
+  (
+    '.create-product input[type="number"]'
+  );
+
+  let cart = document.querySelector("tbody"); 
+
+  let newProduct = document.createElement("tr"); 
+  newProduct.className = "product"; 
+
+  newProduct.innerHTML = `
+  <tr class="product">
```

Comment:
no need to create a new `tr` inside the already existing one. check what innerHTML does.
used it like this, it will create
```
<tr>
  <tr>
    ...other elements here
  </tr>
</tr>
```